### DISCUSSION TITLE
Password genration rules not preserved

### DISCUSSION THREAD

**Discussion by naclander** [dani-garcia/vaultwarden#2412] [Q&A] [answered]
### Subject of the issue
I can successfully change the password generation rules from default ( increasing length from 14 characters to 25 ).
Upon restarting the tab, or when logging in from a different browser, the password generation rules seem to get reverted back to default.

* Install method: Arch Linux package -  Version 2.27.0, sqlite

* Clients used: web

**Comment by BlackDex** [comment, marked as answer]
Those settings are stored into cookies/local browser cache.
You probably have some extension enabled which prevents the storage or clears the storage of these settings.

**Comment by naclander** [comment]
Yes that's most likely it. Is there a way to change the default vaultwarden settings? I looked in `/etc/vaultwarden.env` but did not find anything regarding password generation.

**Comment by BlackDex** [comment]
No, as those settings are as mentioned before per user settings.
The only way to configure a default is if you have an organization for which you can configure some default settings.
https://bitwarden.com/help/policies/#password-generator

But that is about it.

**Reply by naclander** [reply]
Thanks. Since I would like this to be a user setting that can be persisted on the backend, is this worth cutting a new feature ticket for? Or is this more of a bitwarden issue?

And as a follow up, does this mean every new browser I use ( say in other computers ) won't have the password generation preferences synced? So I'll have to keep changing it for every new web client instance? Seems like this is something that should be persisted in the DB, but again, this might be a bitwarden issue not a vaultwarden one.

**Reply by BlackDex** [reply]
This needs to be supported by the clients, Which only checks the cookies/local-storage. So this will probably not work by sending something on the server side without being part of an organization, and that is already there.

So this then would need to be a request for self-hosted environments to support server-side password generation settings without an organization. Which i think is not something they will add, but you may always try of course.

**Reply by naclander** [reply]
Ack, using the Firefox extension is the way I worked around this problem. Thank you.
